feat(fetchye): support headers as a function (#86)
americanexpress/fetchye: README.md

};
```

### Passing dynamic headers

When you call the `run` function returned from useFetchye, it will use the values last rendered into the hook.

This means any correlationId, timestamp, or any other unique dynamic header you might want sent to the server will use its previous value.

To overcome this, you can specify a function instead of a `headers` object in the options.

This function will be called, to re-make the headers just before an API call is made, even when you call `run`.

> Note: You still need to remove dynamic keys from the options using `mapOptionsToKey` otherwise these dynamic headers will bust your cache!

Reviewer: Clunky suggestion below, but maybe phase in a way that does not imply that dynamic headers should always be removed from the cache key

```suggestion
Note: If you don't want the dynamic headers to result in a cache miss remove the keys from the options using `mapOptionsToKey`.
```

Author: ```suggestion
Note: If you don't want the dynamic headers to result in a cache miss, you must remove the keys of the dynamic headers from the options using `mapOptionsToKey` (see example below).
```

Author: @JAdshead I went even more verbose to be extra clear. Lmk your thoughts
